perf/small_alloc_perf.c

```diff
@@ -100,46 +139,79 @@ small_is_unused(void)
 	return true;
 }
 
-static void
-small_alloc_test(int size_min, int size_max, int iterations_max,
-	int rnd, int cnt)
+/*
+ * Function for testing the performance of
+ * memory allocation and deallocation. Return
+ * the total number of operations performed.
+ *
+ * @param[in] size_min - minimal object size for memory allocation
+ * @param[in] size_max - maximal object size for memory allocation
+ * @param[in] iterations_max - count of test iterations
+ * @param[in] rnd - indicates how the object size is calculated
+ *                  (randomly or not)
+ * @param[in] cnt - count of allocated objects
+ */
+static unsigned long long
+small_alloc_test(const int size_min, const int size_max,
+		 const int iterations_max, const int rnd, const int cnt)
 {
+	unsigned long long count = 0;
 	double pow_factor = calculate_pow_factor(size_max, cnt, 256);
+	small_alloc_setopt(&alloc, SMALL_DELAYED_FREE_MODE, false);
+	allocating = true;
+
 	for (int i = 0; i <= iterations_max; i++) {
-		int mode = i % 3;
-		switch (mode) {
-		case 1:
-			small_alloc_setopt(&alloc,
-					   SMALL_DELAYED_FREE_MODE, false);
-			break;
-		case 2:
-			small_alloc_setopt(&alloc,
-					   SMALL_DELAYED_FREE_MODE, true);
-			break;
-		default:
-			break;
+		if (!no_delayed_free_mode) {
+			int mode = i % 3;
+			switch (mode) {
+			case 1:
+				small_alloc_setopt(&alloc,
```

Comment:
Don't get this part. If we have no_delayed_free_mode, why do we have to set it anyway once per three iterations?

Reply:
If it is still required, could you please use enum for mode values (to make it a bit comprehensible).